src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Pack.targets

```diff
@@ -263,9 +263,48 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
     <!-- Include PackageFiles and Content of the project being packed -->
     <ItemGroup>
-      <_PackageFiles Include="@(Content)" Condition=" %(Content.Pack) != 'false' " />
-      <_PackageFiles Include="@(Compile)" Condition=" %(Compile.Pack) == 'true' " />
-      <_PackageFiles Include="@(None)" Condition=" %(None.Pack) == 'true' " />
+      <_PackageFiles Include="@(Content)" Condition=" %(Content.Pack) != 'false' ">
+        <BuildAction>Content</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(Compile)" Condition=" %(Compile.Pack) == 'true' ">
+        <BuildAction>Compile</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(None)" Condition=" %(None.Pack) == 'true' ">
+        <BuildAction>None</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(EmbeddedResource)" Condition=" %(EmbeddedResource.Pack) == 'true' ">
+        <BuildAction>EmbeddedResource</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(ApplicationDefinition)" Condition=" %(ApplicationDefinition.Pack) == 'true' ">
+        <BuildAction>ApplicationDefinition</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(Page)" Condition=" %(Page.Pack) == 'true' ">
+        <BuildAction>Page</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(Resource)" Condition=" %(Resource.Pack) == 'true' ">
+        <BuildAction>Resource</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(SplashScreen)" Condition=" %(SplashScreen.Pack) == 'true' ">
+        <BuildAction>SplashScreen</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(DesignData)" Condition=" %(DesignData.Pack) == 'true' ">
+        <BuildAction>DesignData</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(DesignDataWithDesignTimeCreatableTypes)" Condition=" %(DesignDataWithDesignTimeCreatableTypes.Pack) == 'true' ">
+        <BuildAction>DesignDataWithDesignTimeCreatableTypes</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(CodeAnalysisDictionary)" Condition=" %(CodeAnalysisDictionary.Pack) == 'true' ">
+        <BuildAction>CodeAnalysisDictionary</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(AndroidAsset)" Condition=" %(AndroidAsset.Pack) == 'true' ">
+        <BuildAction>AndroidAsset</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(AndroidResource)" Condition=" %(AndroidResource.Pack) == 'true' ">
+        <BuildAction>AndroidResource</BuildAction>
+      </_PackageFiles>
+      <_PackageFiles Include="@(BundleResource)" Condition=" %(BundleResource.Pack) == 'true' ">
```

Comment:
We need some end to end tests that run this MSBuild file. @mishra14 is working on something like this.